src/map/mail.cpp

```diff
@@ -289,7 +290,24 @@ void mail_getattachment(struct map_session_data* sd, struct mail_message* msg, i
 
 	for( i = 0; i < MAIL_MAX_ITEM; i++ ){
 		if( item->nameid > 0 && item->amount > 0 ){
-			pc_additem(sd, &item[i], item[i].amount, LOG_TYPE_MAIL);
+			// If no card or special card id is set
+			if( item[i].card[0] == 0 ){
+				// Check if it is a pet egg
+				std::shared_ptr<s_pet_db> pet = pet_db_search( item[i].nameid, PET_EGG );
+
+				// If it is a pet egg and the card data does not contain a pet id (see if clause above)
+				if( pet != nullptr ){
+					// Create a new pet
+					pet_create_egg( sd, item[i].nameid );
+				}else{
+					// Add the item normally
+					pc_additem( sd, &item[i], item[i].amount, LOG_TYPE_MAIL );
```

Comment:
Could you merge the 2 elses into one?

Reply:
This was a performance decission so that the pet lookup is only done in this special case and not everytime.

Reply:
I made use of the for loop to merge them as you wanted.

Reply:
Yeah that's what I thought. I was wondering if both things could be achieved, but nothing came to my mind, just asking in case you could find a better solution XD